src/python/nxt_python_wsgi.c

```diff
@@ -604,8 +612,29 @@ nxt_python_get_environ(nxt_python_ctx_t *pctx)
                            r->target_length));
     RC(nxt_python_add_sptr(pctx, nxt_py_query_string_str, &r->query,
                            r->query_length));
-    RC(nxt_python_add_sptr(pctx, nxt_py_path_info_str, &r->path,
-                           r->path_length));
+    if (prefix.length > 0) {
+        i = prefix.length;
+        vl = r->path_length;
+        path = nxt_unit_sptr_get(&r->path);
+        if (vl >= i && nxt_strncmp(prefix.start, path, i) == 0) {
+            RC(nxt_python_add_char(pctx, nxt_py_script_name,
+                                   (char*)prefix.start, i - 1));
+            RC(nxt_python_add_char(pctx, nxt_py_path_info_str,
+                                   path + i - 1, vl - i + 1));
+        }
+        else if ((vl + 1) == i && nxt_strncmp(prefix.start, path, i - 1) == 0) {
+            // Request path matches prefix but doesn't end with "/"
+            RC(nxt_python_add_char(pctx, nxt_py_script_name,
+                                   (char*)prefix.start, i - 1));
+            RC(nxt_python_add_char(pctx, nxt_py_path_info_str, path, 0));
+        }
+        else {
+            RC(nxt_python_add_char(pctx, nxt_py_path_info_str, path, vl));
+        }
+    } else {
+        RC(nxt_python_add_sptr(pctx, nxt_py_path_info_str, &r->path,
+                               r->path_length));
+    }
```

Comment:
It seems that if you do the opposite: always remove "/" from the "prefix" value, then code will look much simpler.

Assuming that "prefix" is always without "/" at the end:
```c
    path = nxt_unit_sptr_get(&r->path);
    path_len = r->path_length;

    if (prefix->length > 0
        && ((path_len > prefix->length && path[prefix->length] == '/')
            || path_len == prefix->length)
        && nxt_memcmp(prefix->start, path, prefix->length) == 0)
    {
        RC(nxt_python_add_char(pctx, nxt_py_script_name,
                               prefix->start, prefix->length));

        path += prefix->length;
        path_len -= prefix->length;
    }

    RC(nxt_python_add_char(pctx, nxt_py_path_info_str, path, path_len));
```